Microsoft.WindowsAzure.Storage/includes/was/service_client.h

```diff
@@ -74,11 +74,22 @@ namespace azure { namespace storage {
         /// Gets the storage account credentials for the service client.
         /// </summary>
         /// <returns>The storage account credentials for the service client.</returns>
-        const azure::storage::storage_credentials& credentials() const
+        const azure::storage::storage_credentials& credentials()
```

Comment:
This is not thread safe. After this function returns and the mutex is released, you still have a const reference to the `storage_credential`, with which you can still read the credential.